src/main/java/org/truffleruby/parser/BodyTranslator.java

```diff
@@ -1757,24 +1757,24 @@ private RubyNode readGlobal(ParseNode node, String variableName) {
             final char type = name.charAt(1);
             switch (type) {
                 case '`':
-                    index = ReadMatchReferenceNode.PRE;
+                    index = ReadMatchReferenceNodes.PRE;
                     break;
                 case '\'':
-                    index = ReadMatchReferenceNode.POST;
+                    index = ReadMatchReferenceNodes.POST;
                     break;
                 case '&':
-                    index = ReadMatchReferenceNode.GLOBAL;
+                    index = ReadMatchReferenceNodes.GLOBAL;
                     break;
                 case '+':
-                    index = ReadMatchReferenceNode.HIGHEST;
+                    index = ReadMatchReferenceNodes.HIGHEST;
```

Comment:
This indirection though constants could be removed now, and we could instantiate ReadHighestMatchNode directly here.